swiftnav/src/reference_frame.rs

```diff
@@ -0,0 +1,656 @@
+use crate::coords::{Coordinate, ECEF};
+use std::{convert::TryFrom, fmt, str::FromStr};
+
+/// Reference Frames
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub enum ReferenceFrame {
+    ITRF2008,
+    ITRF2014,
+    ITRF2020,
+    ETRF2008,
+    ETRF2014,
+    ETRF2020,
+    /// i.e. NAD83(2011)
+    NAD83_2011,
+    /// i.e. NAD83(CSRS) - Canadian Spatial Reference System
+    #[allow(non_camel_case_types)]
+    NAD83_CSRS,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub struct InvalidReferenceFrameName(String);
+
+impl fmt::Display for InvalidReferenceFrameName {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Invalid reference frame name ({})", self.0)
```

Comment:
nit. I'm not sure how frequently you imagine users will be attempting to compile a reference frame from a string. If you think it will be a common usecase, there are some crates which allow you to get a vector of all enum variants. Like https://crates.io/crates/enum-iterator

Then you could iterate over the vector of enum variants to print which reference frames are available whenever this error is hit.

Reply:
I like the strum crate which has an enum iterator and a EnumString macro which would auto-derive the try-from later.

Reply:
I'll look into these crates. I'm usually a little hesitant to pull in more dependencies since it can easily cascade to a huge number, but it seems like that boat as already sailed a bit by us including the rustversion crate which has a pretty large dependency tree of it's own.

I'm going to be really looking for the ability to specify multiple acceptable strings for each variant though, there's a tendency for reference frames to have multiple accepted naming schemes and they aren't always the most logical schemes.

Reply:
I've added strum, seems like you're able to do exactly what I wanted with their attributes